api/tag/schema.py

```diff
@@ -34,5 +35,22 @@ def mutate(self, info, **kwargs):
         return CreateTag(tag=tag)
 
 
+class DeleteTag(graphene.Mutation):
+    class Arguments:
+        tag_id = graphene.Int(required=True)
+    tag = graphene.Field(Tag)
+
+    @Auth.user_roles('Admin')
+    def mutate(self, info, tag_id):
+        query = Tag.get_query(info)
+        tag = query.filter(
+            TagModel.id == tag_id).first()
+        if not tag:
+            raise GraphQLError("Tag not found")
+        tag.delete()
+        return DeleteTag(tag=tag)
```

Comment:
Since we have the soft delete implemented wouldn't it be better if you updated the state instead of deleting the tags?

Reply:
Oh yeah, that's a good point @caveinn. Let me work on it